erpnext/maintenance/doctype/maintenance_schedule/maintenance_schedule.py

```diff
@@ -16,26 +16,25 @@
 class MaintenanceSchedule(TransactionBase):
 	@frappe.whitelist()
 	def generate_schedule(self):
-		self.set('schedules', [])
-		frappe.db.sql("""delete from `tabMaintenance Schedule Detail`
-			where parent=%s""", (self.name))
-		count = 1
-		for d in self.get('items'):
-			self.validate_maintenance_detail()
-			s_list = []
-			s_list = self.create_schedule_list(d.start_date, d.end_date, d.no_of_visits, d.sales_person)
-			for i in range(d.no_of_visits):
-				child = self.append('schedules')
-				child.item_code = d.item_code
-				child.item_name = d.item_name
-				child.scheduled_date = s_list[i].strftime('%Y-%m-%d')
-				if d.serial_no:
-					child.serial_no = d.serial_no
-				child.idx = count
-				count = count + 1
-				child.sales_person = d.sales_person
-				child.completion_status = "Pending"
-				child.item_reference = d.name
+		if not self.docstatus:
```

Comment:
```suggestion
		if self.docstatus != 1:
				return	
```

Always use guard conditions to avoid over indenting.